manim/utils/color/core.py

```diff
@@ -0,0 +1,572 @@
+        value.resize(4, refcheck=False)
+        value[3] = alpha
+        return value
+
+    @staticmethod
+    def _internal_from_int_rgba(rgba: RGBA_Tuple_Int) -> ManimColorInternal:
+        return np.asarray(rgba, dtype=ManimColorDType) / 255
+
+    @staticmethod
+    def _internal_from_rgba(rgba: RGBA_Tuple_Float) -> ManimColorInternal:
+        return np.asarray(rgba, dtype=ManimColorDType)
+
+    # TODO: This may be a bad idea but i don't know what else will be better without writing an endless list of colors
+    @staticmethod
+    def _internal_from_string(name: str) -> ManimColorInternal:
+        from . import _all_color_dict
+
+        if name.upper() in _all_color_dict:
+            return _all_color_dict[name]._internal_value
+        else:
+            raise ValueError(f"Color {name} not found")
+
+    def to_integer(self) -> int:
+        return int.from_bytes(
+            (self._internal_value[:3] * 255).astype(int).tobytes(), "big"
+        )
+
+    def to_rgb(self) -> RGB_Array_Float:
+        return self._internal_value[:3]
+
+    def to_int_rgb(self) -> RGB_Array_Int:
+        return (self._internal_value[:3] * 255).astype(int)
+
+    def to_rgba(self) -> RGBA_Array_Float:
+        return self._internal_value
+
+    def to_int_rgba(self) -> RGBA_Array_Int:
+        return (self._internal_value * 255).astype(int)
+
+    def to_rgba_with_alpha(self, alpha: float) -> RGBA_Array_Float:
+        return np.fromiter((*self._internal_value[:3], alpha), dtype=ManimColorDType)
+
+    # @deprecated("Use to_rgb_with_alpha instead.")
+    def to_int_rgba_with_alpha(self, alpha: float) -> RGBA_Array_Int:
+        tmp = self._internal_value * 255
+        tmp[3] = alpha * 255
+        return tmp.astype(int)
+
+    def to_hex(self, with_alpha: bool = False) -> str:
+        tmp = f"#{int(self._internal_value[0]*255):02X}{int(self._internal_value[1]*255):02X}{int(self._internal_value[2]*255):02X}"
+        if with_alpha:
+            tmp += f"{int(self._internal_value[3]*255):02X}"
+        return tmp
+
+    def invert(self, with_alpha=False) -> ManimColor:
+        return ManimColor(1.0 - self._internal_value, with_alpha)
+
+    def interpolate(self, other: ManimColor, alpha: float) -> ManimColor:
+        return ManimColor(
+            self._internal_value * (1 - alpha) + other._internal_value * alpha
+        )
+
+    @classmethod
+    def from_rgb(
+        cls,
+        rgb: RGB_Array_Float | RGB_Tuple_Float | RGB_Array_Int | RGB_Tuple_Int,
+        alpha: float = 1.0,
+    ) -> ManimColor:
+        return cls(rgb, alpha)
+
+    @classmethod
+    def from_rgba(
+        cls, rgba: RGBA_Array_Float | RGBA_Tuple_Float | RGBA_Array_Int | RGBA_Tuple_Int
+    ) -> ManimColor:
+        return cls(rgba)
+
+    @classmethod
+    def from_hex(cls, hex: str, alpha: float = 1.0) -> ManimColor:
+        return cls(hex, alpha)
+
+    @classmethod
+    def parse(
+        cls,
+        color: ParsableManimColor | list[ParsableManimColor] | None,
```

Comment:
Why not use `Sequence` instead of `list`? Tuples seems to be allowed (+ in a bunch of other places)